### PR TITLE
Move the task runner to ESM

### PR DESCRIPTION
**Pull request by mxdvl**
## What is the value of this and can you measure success?

Enable bumping ESM-only dependencies of the task runner, such as `cpy`

## What does this change?

Refactor all the CommonJS of the task runner to use ESM instead.

> [!NOTE]
> This does not yet use ESM for the tasks themselves

## Screenshots

image

## Checklist

- [X] Tested locally, and on CODE if necessary
- [X] Will not break dotcom-rendering
- [X] Will not break our database – if updating CAPI, [updated and committed the database files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### CODE REVIEW

File: static/public/images/email/banners/hear-here.gif
Comment (domlander) [inline static/public/images/email/banners/hear-here.gif:1]: What's the change here?

Comment (mxdvl) [reply static/public/images/email/banners/hear-here.gif:1]: an error!

**Comment by mxdvl** [comment]
Note:

```js
/** @type {import('../../task-runner/task.mjs').Task} */
/** @typedef {{
    description: string;
    task: Array<() => Promise<void> | Task>;
    concurrent?: boolean;
}} Task */

```

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,11 +34,11 @@ jobs:
...
-        run: ./tools/task-runner/runner validate test
+      - run: make validate
 
-      - name: assets compile
-        run: ./tools/task-runner/runner compile
+      - run: make test
+
+      - run: make compile
         env:
           NODE_ENV: production
```
Comment (mxdvl) [inline .github/workflows/build.yml:43]: This is redundant!

```suggestion
```

**Review by sunnyclimber456 — APPROVED** [review]

**Pull request merged**